### ISSUE TITLE
Folding No Longer Working

### ISSUE THREAD

**Issue by iohcidnal** [Microsoft/vscode#47454]
Issue Type: <b>Bug</b>

Opening a React js file no longer displays folding controls on mouse hover in the render() function for components. For example:
```
      <PageHeader title="Log in"   // this does not show the fold control
        {...{  // this one does
          item: true,
          xs: 9,
          md: 5
        }}
      />
```

VS Code version: Code 1.22.1 (950b8b0d37a9b7061b6f0d291837ccc4015f5ecd, 2018-04-06T00:21:13.607Z)
OS version: Darwin x64 17.4.0

<details><summary>Extensions (27)</summary>

Extension|Author (truncated)|Version
---|---|---
bracket-pair-colorizer|Coe|1.0.27
vscode-jest|Ort|2.6.4
vscode-docker|Pet|0.0.26
quokka-vscode|Wal|1.0.117
html-css-class-completion|Zig|1.17.1
html-snippets|abu|0.2.1
react-pure-to-class-vscode|ang|1.1.4
cake-vscode|cak|0.10.1
jsrefactor|cms|2.11.6
vscode-eslint|dba|1.4.8
gitlens|eam|8.2.0
tslint|eg2|1.0.28
auto-close-tag|for|0.5.6
auto-rename-tag|for|0.0.15
csharpextensions|jch|1.3.0
sharppad|jma|1.0.5
mssql|ms-|1.3.0
azure-account|ms-|0.4.0
csharp|ms-|1.14.0
PowerShell|ms-|1.6.0
debugger-for-chrome|msj|4.3.0
syncing|non|1.5.3
convert-css-in-js|pau|1.0.0
vscode-icons|rob|7.22.0
guides|spy|0.9.1
vscode-todo-highlight|way|0.5.12
JavaScriptSnippets|xab|1.5.0

</details>

**Comment by btkfrank** [comment]
same here in the recent update

**Comment by btkfrank** [comment]
[https://github.com/Microsoft/vscode/issues/46989](url)
This works for me

**Comment by iohcidnal** [comment]
@btkfrank I'm getting 404 from the link you posted.

**Comment by chatdanai** [comment]
Same on my machine, OSX. It's no longer work in JSX section. typescript section is fine.

**Comment by shobhitchittora** [comment]
+1 on this. 👍🏻

**Comment by mjbvz** [comment]
Root caused by https://github.com/Microsoft/TypeScript/issues/23273

Will be fixed by #47473

**Comment by btkfrank** [comment]
@iohcidnal for now you can temporarily add this to your user settings(cmd + ',' on mac)

`"editor.foldingStrategy": "indentation"`

Otherwise, wait for the new release
